azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/service/process/AzureProcessService.java

```diff
@@ -0,0 +1,74 @@
+import com.synopsys.integration.azure.boards.common.model.AzureArrayResponseModel;
+import com.synopsys.integration.azure.boards.common.util.AzureSpecTemplate;
+
+/**
+ * <a href="https://docs.microsoft.com/en-us/rest/api/azure/devops/processes/work%20item%20types/list?view=azure-devops-rest-5.1">Work Item Types</a>
+ * <a href="https://docs.microsoft.com/en-us/rest/api/azure/devops/processes/fields/add?view=azure-devops-rest-5.1">Fields</a>
+ */
+public class AzureProcessService {
+    public static final AzureSpecTemplate API_SPEC_ORGANIZATION_PROCESS_WORKITEMTYPES = new AzureSpecTemplate("/{organization}/_apis/work/processes/{processId}/workItemTypes");
+    public static final AzureSpecTemplate API_SPEC_ORGANIZATION_PROCESS_WORKITEMTYPE_FIELDS = new AzureSpecTemplate("/{organization}/_apis/work/processes/{processId}/workItemTypes/{witRefName}/fields");
+    public static final String PATH_ORGANIZATION_REPLACEMENT = "{organization}";
+    public static final String PATH_PROCESS_ID_REPLACEMENT = "{processId}";
+
+    private final AzureHttpService azureHttpService;
+
+    public AzureProcessService(AzureHttpService azureHttpService) {
+        this.azureHttpService = azureHttpService;
+    }
+
+    public AzureArrayResponseModel<ProcessWorkItemTypesResponseModel> getWorkItemTypes(String organizationName, String processId) throws HttpServiceException {
+        String requestSpec = API_SPEC_ORGANIZATION_PROCESS_WORKITEMTYPES
+                                 .defineReplacement(PATH_ORGANIZATION_REPLACEMENT, organizationName)
+                                 .defineReplacement(PATH_PROCESS_ID_REPLACEMENT, processId)
+                                 .populateSpec();
+        requestSpec = appendApiVersionQueryParam(requestSpec);
+        Type responseType = new TypeToken<AzureArrayResponseModel<ProcessWorkItemTypesResponseModel>>() {}.getType();
+        return azureHttpService.get(requestSpec, responseType);
+    }
+
+    public ProcessFieldResponseModel addFieldToWorkItemType(String organizationName, String processId, String workItemTypeRefName, ProcessFieldRequestModel requestBody) throws IOException, HttpServiceException {
+        String requestSpec = API_SPEC_ORGANIZATION_PROCESS_WORKITEMTYPE_FIELDS
+                                 .defineReplacement(PATH_ORGANIZATION_REPLACEMENT, organizationName)
+                                 .defineReplacement(PATH_PROCESS_ID_REPLACEMENT, processId)
+                                 .defineReplacement("{witRefName}", workItemTypeRefName)
+                                 .populateSpec();
+        requestSpec = appendApiVersionQueryParam(requestSpec);
+        return azureHttpService.post(requestSpec, requestBody, ProcessFieldResponseModel.class);
+    }
+
+    private String appendApiVersionQueryParam(String requestSpec) {
+        return String.format("%s?%s=%s", requestSpec, AzureHttpService.AZURE_API_VERSION_QUERY_PARAM_NAME, "5.1-preview.2");
```

Comment:
Why not put the 3rd variable directly in the string?

Reply:
And if it's important that it's separate, then we should make it into a variable